src/proc/hdr-merge.cpp

```diff
@@ -258,6 +258,13 @@ namespace librealsense
         // checking frame counter of first depth and ir are the same
         if (use_ir)
         {
+            // on devices that does not support meta data on IR frames, do not use IR for hdr merging
```

Comment:
I must say this this function is written very badly :)
But I know it's not your code ..